systemverilog/SystemVerilog.g4

```diff
@@ -0,0 +1,3757 @@
+// Author: Mustafa Said Ağca
+// License: MIT
+
+grammar SystemVerilog;
+import SystemVerilogLexerRules;
+
+/*******************
+	PARSER RULES
+*******************/
+
+// A.1.1 Library source text
+
+/*library_text
+	: (library_description)*
+	;
+
+library_description
+	: library_declaration
+	| include_statement
+	| config_declaration
+	| ';'
+	;
+
+library_declaration
+	: 'library' library_identifier FILE_PATH_SPEC (',' FILE_PATH_SPEC)* ('-incdir' FILE_PATH_SPEC (',' FILE_PATH_SPEC)*)? ';'
+	;
+
+include_statement
+	: 'include' FILE_PATH_SPEC ';'
+	;
+*/
+// A.1.2 SystemVerilog source text
+
+source_text
+	: (timeunits_declaration)? (description)* EOF
+	;
+
+description
+	: module_declaration
+	//| udp_declaration
+	| interface_declaration
+	| program_declaration
+	| package_declaration
+	| (attribute_instance)* package_item
+	| (attribute_instance)* bind_directive
+	//| config_declaration
+	;
+
+module_nonansi_header
+	: (attribute_instance)* module_keyword (lifetime)? module_identifier (package_import_declaration)* (parameter_port_list)? list_of_ports ';'
+	;
+
+module_ansi_header
+	: (attribute_instance)* module_keyword (lifetime)? module_identifier (package_import_declaration)* (parameter_port_list)? (list_of_port_declarations)? ';'
+	;
+
+module_declaration
+	: module_nonansi_header (timeunits_declaration)? (module_item)* 'endmodule' (':' module_identifier)?
+	| module_ansi_header (timeunits_declaration)? (non_port_module_item)* 'endmodule' (':' module_identifier)?
+	| (attribute_instance)* module_keyword (lifetime)? module_identifier '(' '.*' ')' ';' (timeunits_declaration)? (module_item)* 'endmodule' (':' module_identifier)?
+	| 'extern' module_nonansi_header
+	| 'extern' module_ansi_header
```

Comment:
Such alternatives can be merged. It's good for clarity and performance:

```suggestion
	| 'extern' (module_nonansi_header | module_ansi_header)
```

Here and in entire grammar.